.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,36 @@
+name: Build docs
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - docs
+  workflow_dispatch:
+  
+jobs:
+
+  build-and-upload-artifact:
```

Comment:
Can rename this to just "build"? There's no artifact uploading anywhere as far as I can tell? Or (ideally) add an upload artifact step at the end.